wdl_runner/cromwell_launcher/cromwell_driver.py

```diff
@@ -110,12 +114,28 @@ def submit(self, wdl, workflow_inputs, workflow_options, sleep_time=15):
 
     # Job is running.
     cromwell_id = job['id']
-    logging.info("Cromwell job id: %s", cromwell_id)
+    logging.info("Job submitted to Cromwell. job id: %s", cromwell_id)
 
-    # Poll for completion.
+    # Poll Cromwell for job completion.
+    attempt = 0
     while True:
       time.sleep(sleep_time)
-      status_json = self.fetch(wf_id=cromwell_id, method='status')
+
+      # Cromwell occassionally fails to respond to the status request.
+      # Only give up after 3 consecutive failed requests.
+      try:
+        status_json = self.fetch(wf_id=cromwell_id, method='status')
+        attempt = 0
+      except requests.exceptions.ConnectionError as e:
+        attempt += 1
+        logging.info("Error polling Cromwell job status (attempt %d): %s",
+          attempt, e)
+
+        if attempt >= 3:
```

Comment:
And it probably also should be `attempt == max_num_retries` since it checks it after every failed attempt, so it will never be more than 3 failed attempts before it exits.